# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
# Commit: 09db50c8f370
# Full Hash: 09db50c8f370b69f186f3e6b7ce6154a984b40b4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Pass codec configs by value/reference instead of pointers. r=ng
#   
#   This patch mainly sets the stage for using codec configs with StateMirroring.
#   It also makes the code a bit simpler, and removes the need to handle the nullptr
#   case.
# ==============================================================================

diff -r d1552c620c11 -r 09db50c8f370 dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
--- a/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Wed Aug 11 11:33:21 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Wed Apr 14 17:36:40 2021 +0200
@@ -408,7 +408,7 @@
    *
    */
   virtual MediaConduitErrorCode ConfigureSendMediaCodec(
-      const VideoCodecConfig* sendSessionConfig,
+      const VideoCodecConfig& sendSessionConfig,
       const RtpRtcpConfig& aRtpRtcpConfig) = 0;
 
   /**
@@ -419,7 +419,7 @@
    *
    */
   virtual MediaConduitErrorCode ConfigureRecvMediaCodecs(
-      const std::vector<UniquePtr<VideoCodecConfig>>& recvCodecConfigList,
+      const std::vector<VideoCodecConfig>& recvCodecConfigList,
       const RtpRtcpConfig& aRtpRtcpConfig) = 0;
 
   /**
@@ -524,7 +524,7 @@
    * NOTE: See VideoConduit for more information
    */
   virtual MediaConduitErrorCode ConfigureSendMediaCodec(
-      const AudioCodecConfig* sendCodecConfig) = 0;
+      const AudioCodecConfig& sendCodecConfig) = 0;
 
   /**
    * Function to configure list of receive codecs for the audio session
@@ -532,7 +532,7 @@
    * NOTE: See VideoConduit for more information
    */
   virtual MediaConduitErrorCode ConfigureRecvMediaCodecs(
-      const std::vector<UniquePtr<AudioCodecConfig>>& recvCodecConfigList) = 0;
+      const std::vector<AudioCodecConfig>& recvCodecConfigList) = 0;
 
   virtual bool SetDtmfPayloadType(unsigned char type, int freq) = 0;
 